samza-core/src/main/java/org/apache/samza/execution/JobNode.java

```diff
@@ -92,14 +93,21 @@ void addOutEdge(StreamEdge out) {
     return outEdges;
   }
 
-  public JobConfig generateConfig() {
+  /**
+   * Generate the configs for a job
+   * @param executionPlanJson JSON representation of the execution plan
+   * @return config of the job
+   */
+  public JobConfig generateConfig(String executionPlanJson) {
```

Comment:
sure. Changed the config name to be samza.internal.execution.plan.

For jobGraph it will be ok. But it might cause issue for operator lamdas. I will send a discussion email about it once the feature is onboard.